test/kb_blast_server_test.py

```diff
@@ -126,24 +249,128 @@ def test_kb_blast_BLASTn_Search_01(self):
+                                                                }
+                                                            ]
+                                                        }]
+                                                })[0]
+
+        #pprint(obj_info)
+        target_genomeSet_ref = "/".join([str(obj_info[WORKSPACE_I]),
+                                         str(obj_info[OBJID_I]),
+                                         str(obj_info[VERSION_I])])
+
+
+        # E. coli K-12 MG1655 dnaA
+        query_seq_nuc = 'GTGTCACTTTCGCTTTGGCAGCAGTGTCTTGCCCGATTGCAGGATGAGTTACCAGCCACAGAATTCAGTATGTGGATACGCCCATTGCAGGCGGAACTGAGCGATAACACGCTGGCCCTGTACGCGCCAAACCGTTTTGTCCTCGATTGGGTACGGGACAAGTACCTTAATAATATCAATGGACTGCTAACCAGTTTCTGCGGAGCGGATGCCCCACAGCTGCGTTTTGAAGTCGGCACCAAACCGGTGACGCAAACGCCACAAGCGGCAGTGACGAGCAACGTCGCGGCCCCTGCACAGGTGGCGCAAACGCAGCCGCAACGTGCTGCGCCTTCTACGCGCTCAGGTTGGGATAACGTCCCGGCCCCGGCAGAACCGACCTATCGTTCTAACGTAAACGTCAAACACACGTTTGATAACTTCGTTGAAGGTAAATCTAACCAACTGGCGCGCGCGGCGGCTCGCCAGGTGGCGGATAACCCTGGCGGTGCCTATAACCCGTTGTTCCTTTATGGCGGCACGGGTCTGGGTAAAACTCACCTGCTGCATGCGGTGGGTAACGGCATTATGGCGCGCAAGCCGAATGCCAAAGTGGTTTATATGCACTCCGAGCGCTTTGTTCAGGACATGGTTAAAGCCCTGCAAAACAACGCGATCGAAGAGTTTAAACGCTACTACCGTTCCGTAGATGCACTGCTGATCGACGATATTCAGTTTTTTGCTAATAAAGAACGATCTCAGGAAGAGTTTTTCCACACCTTCAACGCCCTGCTGGAAGGTAATCAACAGATCATTCTCACCTCGGATCGCTATCCGAAAGAGATCAACGGCGTTGAGGATCGTTTGAAATCCCGCTTCGGTTGGGGACTGACTGTGGCGATCGAACCGCCAGAGCTGGAAACCCGTGTGGCGATCCTGATGAAAAAGGCCGACGAAAACGACATTCGTTTGCCGGGCGAAGTGGCGTTCTTTATCGCCAAGCGTCTACGATCTAACGTACGTGAGCTGGAAGGGGCGCTGAACCGCGTCATTGCCAATGCCAACTTTACCGGACGGGCGATCACCATCGACTTCGTGCGTGAGGCGCTGCGCGACTTGCTGGCATTGCAGGAAAAACTGGTCACCATCGACAATATTCAGAAGACGGTGGCGGAGTACTACAAGATCAAAGTCGCGGATCTCCTTTCCAAGCGTCGATCCCGCTCGGTGGCGCGTCCGCGCCAGATGGCGATGGCGCTGGCGAAAGAGCTGACTAACCACAGTCTGCCGGAGATTGGCGATGCGTTTGGTGGCCGTGACCACACGACGGTGCTTCATGCCTGCCGTAAGATCGAGCAGTTGCGTGAAGAGAGCCACGATATCAAAGAAGATTTTTCAAATTTAATCAGAACATTGTCATCGTAA'
+
+        parameters = { 'workspace_name': self.getWsName(),
+                       'input_one_sequence': query_seq_nuc,
+                       #'input_one_ref': "",
+                       'output_one_name': obj_basename+'.'+"test_query.SS",
+                       'input_many_refs': [target_genomeSet_ref],
+                       'output_filtered_name': obj_out_name,
+                       'genome_disp_name_config': 'obj_name',
+                       #'e_value': ".001",
+                       #'bitscore': "50",
+                       #'ident_thresh': "10.0",
+                       #'overlap_fraction': "50.0",
+                       'e_value': ".1",
+                       'bitscore': "10",
+                       'ident_thresh': "10.0",
+                       'overlap_fraction': "10.0",
+                       'maxaccepts': "1000",
+                       'output_extra_format': "none"
+                     }
+
+        ret = self.getImpl().BLASTn_Search(self.getContext(), parameters)[0]
```

Comment:
It's better not to routinely run wrapped apps as part of the tests -- apart from anything else, it makes the tests a lot slower than they need to be. Better ways to approach this might be
- mock out the part of the code where the blast executable is called, and make sure that various combinations of input params get translated into the correct command line params;
- structure the code such that you can easily mock things like calls to the workspace and replace them with known data

Ideally there would be very few tests that actually run the executable, but lots of tests to ensure that the guts of the KBase app wrapper are working correctly and handle errors appropriately.

Reply:
Frankly in the early days we weren't really using the unit tests as unit tests, rather as integration tests to make sure a given set of parameters and data would run an App through to completion.  That's still true today, as we don't really have enough time to do integration testing, so at least having the "unit tests" do it is something.  We do need more integration testing (and unit testing frankly).  That said, I don't have time now to work on these tests for the BLAST module anymore because I have lots more on my plate to get to.  As long as these integration tests are passing, and they are, that's the best I can do for now.  I do appreciate your point, and wish we had the cycles for more rigor.